### PR TITLE
Anglerecon

### PR DESCRIPTION
**Pull request by gvolpe79**

### CODE REVIEW

File: DataFormats/Reconstruction/include/ReconstructionDataFormats/MatchInfoHMP.h [hunk 1]
```diff
@@ -54,18 +74,67 @@ class MatchInfoHMP
...
+  float mTrkY;           // local y coordinate of extrapolated track intersection point
+  float mTrkTheta;       // theta track
+  float mTrkPhi;         // phi track
   float mCkovAngle;      // emission angle value
-  float mMipCluQ = 0.0;  // MIP cluster charge
   int mMipCluSize = 0.0; // MIP cluster size
   int mNPhots = 0.0;     // number of candidate photo-electrons
   int mIdxPhotClus;      // index of the first photo
+  int mHMPqn;            // 1000000*number of photon clusters + QDC
+  float mPhotCharge[10];
```
Comment (shahor02) [inline DataFormats/Reconstruction/include/ReconstructionDataFormats/MatchInfoHMP.h]: init array as `float mPhotCharge[10] = {};`

File: DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackHMP.h [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  ~TrackHMP() = default;
+
+  Bool_t intersect(Double_t pnt[3], Double_t norm[3], double bz) const; //
+  void propagate(Double_t len, std::array<float, 3>& x, std::array<float, 3>& p, double bz) const;
+
+  // protected:
+  //    Bool_t   Update(const AliCluster */*c*/, Double_t /*chi2*/, Int_t /*idx*/) {return 0;}
+  //    Double_t GetPredictedChi2(const AliCluster */*c*/) const {return 0.;}
+ private:
+  ClassDef(TrackHMP, 0) // HMPID reconstructed tracks
```
Comment (shahor02) [inline DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackHMP.h]: 0 has a special meaning, start from 1

File: Detectors/GlobalTracking/include/GlobalTracking/MatchHMP.h [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+  void setTPCClustersInp(const o2::tpc::ClusterNativeAccess* inp)
+  {
+    mTPCClusterIdxStruct = inp;
+  }
+
+  void setFIT(bool value = true) { mIsFIT = value; }
+  int findFITIndex(int bc);
+
+  ///< populate externally provided container by TOF-time-constrained TPC tracks
+  template <typename V>
```
Comment (shahor02) [inline Detectors/GlobalTracking/include/GlobalTracking/MatchHMP.h]: what is this?

**Review by shahor02 — CHANGES_REQUESTED** [review]

**Comment by gvolpe79** [comment]
Done!
Giacomo.

------------------------
Giacomo Volpe, Ph.D.
Associate Professor - Dipartimento Interateneo di Fisica ‘M. Merlin’of Bari University and Sezione INFN
Via Giovanni Amendola, 173
Bari, ITALY
contacts:
***@***.***, ***@***.***, ***@***.***
tel. +39 5443242
------------------------
________________________________
Da: Ruben Shahoyan ***@***.***>
Inviato: martedì 30 maggio 2023 10:19
A: AliceO2Group/AliceO2 ***@***.***>
Cc: Giacomo Volpe ***@***.***>; Author ***@***.***>
Oggetto: Re: [AliceO2Group/AliceO2] Anglerecon (PR #11400)


@shahor02 requested changes on this pull request.

________________________________

In DataFormats/Reconstruction/include/ReconstructionDataFormats/MatchInfoHMP.h<https://github.com/AliceO2Group/AliceO2/pull/11400#discussion_r1209895826>:

>    int mMipCluSize = 0.0; // MIP cluster size
   int mNPhots = 0.0;     // number of candidate photo-electrons
   int mIdxPhotClus;      // index of the first photo
+  int mHMPqn;            // 1000000*number of photon clusters + QDC
+  float mPhotCharge[10];


init array as float mPhotCharge[10] = {};

________________________________

In DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackHMP.h<https://github.com/AliceO2Group/AliceO2/pull/11400#discussion_r1209897207>:

> + public:
+  TrackHMP();
+  TrackHMP(const TrackHMP& t) = default;
+  TrackHMP(const o2::track::TrackParCov& t) : o2::track::TrackParCov{t} {};
+  //  TrackHMP(const o2::track::TrackParCov& t);
+  TrackHMP& operator=(const o2::track::TrackParCov& t);
+  ~TrackHMP() = default;
+
+  Bool_t intersect(Double_t pnt[3], Double_t norm[3], double bz) const; //
+  void propagate(Double_t len, std::array<float, 3>& x, std::array<float, 3>& p, double bz) const;
+
+  // protected:
+  //    Bool_t   Update(const AliCluster */*c*/, Double_t /*chi2*/, Int_t /*idx*/) {return 0;}
+  //    Double_t GetPredictedChi2(const AliCluster */*c*/) const {return 0.;}
+ private:
+  ClassDef(TrackHMP, 0) // HMPID reconstructed tracks


0 has a special meaning, start from 1

________________________________

In Detectors/GlobalTracking/include/GlobalTracking/MatchHMP.h<https://github.com/AliceO2Group/AliceO2/pull/11400#discussion_r1209899615>:

> +  ///< populate externally provided container by TOF-time-constrained TPC tracks
+  template <typename V>


what is this?

File: DataFormats/Reconstruction/include/ReconstructionDataFormats/MatchInfoHMP.h [hunk 4]
```diff
@@ -28,22 +28,42 @@ class MatchInfoHMP
...
+  void setTrkY(float y) { mTrkY = y; }
+  float getTrkY() const { return mTrkY; }
+
+  void setHMPsignal(float angle) { mCkovAngle = angle; }
+  float getHMPsignal() const
+  {
+    if (mCkovAngle > 0)
+      return mCkovAngle - (Int_t)mCkovAngle;
+    else
+      return mCkovAngle;
```
Comment (shahor02) [inline DataFormats/Reconstruction/include/ReconstructionDataFormats/MatchInfoHMP.h]: please add `{}`, it will trigger CI problems in the following PRs.

**Review by shahor02 — APPROVED** [review]

**Pull request merged**